### ISSUE TITLE
Show version number in CLI output

### ISSUE THREAD

**Issue by warriordog** [warriordog/mooltipage#57]
It should be possible to read this from package.json at runtime and/or build time. Either is fine, as long as the version number is not hard-coded in the output.

**Comment by TvrtkoM** [comment]
hi, im willing to help with this

**Comment by warriordog** [comment]
Hello @TvrtkoM, thanks for the help! I haven't written a contributing guide yet, so please let me know if you have any questions or need a hand with anything.

**Comment by Nicholaiii** [comment]
I took a look at this too, and with the current TypeScript configuration it's really messy, because of the way the paths relate to each other. I can take a deep dive into resolving it if you want, but it might require som restructuring.

**Comment by warriordog** [comment]
@Nicholaiii 
I've also researched this a bit, and I had come up with a solution that shouldn't require too much refactoring. What do you think about this:

Based on [this SO answer](https://stackoverflow.com/a/10855054/1857993), have a file that imports and parses package.json using relative paths. That file then extracts and re-exports the version number as a constant. If TypeScript cannot compile a file using "require", then it should be possible to write this part in JavaScript with a manually-created .d.ts file.

Please feel free to take a look at this if you like, but let me know if you think it might require major changes. If I don't hear anything from @TvrtkoM in the next day or so, then I'll reassign this to you if you're still interested.

**Comment by Nicholaiii** [comment]
I've tried both import and require, and like I pointed out, it's really messy because of the paths. I implemented it exactly like the SO answer specified and it works until the commit guard kicks in and compiles the project in which case it looks for cli.ts/package.json for some odd reason and fails, because cli.ts is not a folder.

**Comment by TvrtkoM** [comment]
I have already cam up with the solution which works.
In `src/lib/fs/fsUtils.ts` added function readPackageJson  (readFileSync + parseJSON) which reads first package.json it finds in one of parent directories and then use it to read and display version number in `src/bin/cli.ts`

require or import, full relative path is needed so `../../../package.json`

I also tried it using https://github.com/npm/pacot which handles things asynchronous for one, and is another dependency so I gave up on it.

I will open working PR today

**Comment by warriordog** [comment]
@Nicholaiii That's really weird, lint-staged should only pass files to tsc and it sounds like its passing a directory.

@TvrtkoM I'm glad to hear that, thanks for your effort! I'll be watching for the PR.

**Comment by TvrtkoM** [comment]
husky git commit hook is broken. It seems it tries to compile every single ts file as a project (passes filename to npm build command - tsc -b [filename])

from package.json
```
"lint-staged": {
    "./**/*.ts": [
      "npm run lint",
      "npm build"
    ]
  }
```

First time seeing lint-staged, but npm build command looks strange. Shouldn't it be `npm run build`. Either way,  the `npm build` script is executed for each file e.g.

`tsc -b src/bin/cli.ts`
which gives error:

`error TS5083: Cannot read file '/home/tvrtko/programming/js/mooltipage/src/bin/cli.ts/tsconfig.json'`

**Comment by TvrtkoM** [comment]
I have included the fix for pre-commit hook on the same branch

**Comment by Nicholaiii** [comment]
This would probably allow you to import or require package.json too.

**Comment by warriordog** [comment]
@TvrtkoM Nice catch with the "build" vs "run build" difference. It looks like `npm build` [is an internal command](https://stackoverflow.com/a/43664456/1857993) that somehow happened to mostly work in this case.

**Comment by warriordog** [comment]
Closed with #70
